lightning/src/ln/functional_test_utils.rs

```diff
@@ -1327,15 +1327,19 @@ macro_rules! expect_payment_path_successful {
 }
 
 macro_rules! expect_payment_forwarded {
-	($node: expr, $source_node: expr, $expected_fee: expr, $upstream_force_closed: expr) => {
+	($node: expr, $source_node: expr, $sink_node: expr, $expected_fee: expr, $upstream_force_closed: expr) => {
 		let events = $node.node.get_and_clear_pending_events();
 		assert_eq!(events.len(), 1);
 		match events[0] {
-			Event::PaymentForwarded { fee_earned_msat, source_channel_id, claim_from_onchain_tx } => {
+			Event::PaymentForwarded { fee_earned_msat, source_channel_id, claim_from_onchain_tx, sink_channel_id } => {
 				assert_eq!(fee_earned_msat, $expected_fee);
 				if fee_earned_msat.is_some() {
-					// Is the event channel_id in one of the channels between the two nodes?
+					// Is the event source_channel_id in one of the channels between the two nodes?
 					assert!($node.node.list_channels().iter().any(|x| x.counterparty.node_id == $source_node.node.get_our_node_id() && x.channel_id == source_channel_id.unwrap()));
+					// After close, channel isn't available
+					if $node.node.list_channels().len() > 1 {
+						assert!($node.node.list_channels().iter().any(|x| x.counterparty.node_id == $sink_node.node.get_our_node_id() && x.channel_id == sink_channel_id.unwrap()));
+					}
```

Comment:
@TheBlueMatt After the channel closes, it isn't available so we should skip it's testing. I've resorted to a length check for now but is there a better way to do this?

Reply:
The tests this was failing at pass if I replace the check with `if !$upstream_force_closed` but i'm not sure why. What does upstream force closing have to do with the sink channel id?